ion-schema/src/types.rs

```diff
@@ -368,12 +405,16 @@ impl TypeDefinitionImpl {
             constraints.push(constraint);
         }
 
+        let isl_struct = &isl_type.isl_type_struct.as_ref();
         // add `type: any` as a default type constraint if there is no type constraint found
         if !found_type_constraint {
             // set the isl type name for any error that is returned while parsing its constraints
             let isl_type_name = match type_name.to_owned() {
                 Some(name) => name,
-                None => "".to_owned(),
+                None => match isl_struct {
+                    None => "".to_owned(),
+                    Some(isl_type_struct) => isl_type_struct.to_owned().to_owned(),
```

Comment:
Is the double `to_owned()` necessary?
```suggestion
                    Some(isl_type_struct) => isl_type_struct.to_owned(),
```

Reply:
No it is not necessary, will remove that